File: tests/selenium/Pages/HomePage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using PolarDrive.Selenium.Tests.Support;

namespace PolarDrive.Selenium.Tests.Pages;

public class HomePage
{
    private readonly IWebDriver _driver;
    private readonly WebDriverWait _wait;

    public NavigationComponent Nav { get; }

    public HomePage(IWebDriver driver)
    {
        _driver = driver;
        _wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10));
        Nav = new NavigationComponent(driver);
    }

    public void Goto(string baseUrl) => _driver.Navigate().GoToUrl(baseUrl);

    private IWebElement HeroCtaButton =>
        _wait.WaitForClickable(By.CssSelector("[data-testid='hero-cta-button']"));

    public void ClickHeroCta() => HeroCtaButton.Click();

    private IWebElement ContactForm =>
        _wait.WaitForVisible(By.CssSelector("[data-testid='contact-form']"));

    private IWebElement NameInput =>
        _driver.FindElement(By.CssSelector("[data-testid='contact-name-input']"));

    private IWebElement EmailInput =>
        _driver.FindElement(By.CssSelector("[data-testid='contact-email-input']"));

    private IWebElement CompanyInput =>
        _driver.FindElement(By.CssSelector("[data-testid='contact-company-input']"));

    private IWebElement MessageTextarea =>
        _driver.FindElement(By.CssSelector("[data-testid='contact-message-textarea']"));

    private IWebElement SubmitButton =>
        _driver.FindElement(By.CssSelector("[data-testid='contact-submit-button']"));

    public string NameInputValue => NameInput.GetAttribute("value") ?? string.Empty;

    public string MessageInputValue => MessageTextarea.GetAttribute("value") ?? string.Empty;

    public void ScrollToContactForm()
    {
        var element = ContactForm;
        ((IJavaScriptExecutor)_driver).ExecuteScript(
            "arguments[0].scrollIntoView({ block: 'center' });", element);
    }

    // Selenium non ha un'API nativa di network interception paragonabile a
    // Playwright's page.route(): qui stubbiamo window.fetch via JS injection
    // per evitare di colpire il vero endpoint Google Apps Script durante il test.
    public void StubFetchForContactForm()
    {
        const string script = """
            window.fetch = function () {
                return Promise.resolve({ ok: true, status: 200, json: () => Promise.resolve({}) });
            };
            """;
        ((IJavaScriptExecutor)_driver).ExecuteScript(script);
    }

    public void FillContactForm(string name, string email, string message, string? company = null)
    {
        NameInput.Clear();
        NameInput.SendKeys(name);

        EmailInput.Clear();
        EmailInput.SendKeys(email);

        if (!string.IsNullOrEmpty(company))
        {
            CompanyInput.Clear();
            CompanyInput.SendKeys(company);
        }

        MessageTextarea.Clear();
        MessageTextarea.SendKeys(message);
    }

    public void SubmitContactForm() => SubmitButton.Click();
}

File: tests/playwright/pages/HomePage.ts
import type { Locator, Page } from "@playwright/test";
import { NavigationComponent } from "./NavigationComponent";

export class HomePage {
  readonly page: Page;
  readonly nav: NavigationComponent;
  readonly heroCtaButton: Locator;

  readonly contactForm: Locator;
  readonly contactNameInput: Locator;
  readonly contactEmailInput: Locator;
  readonly contactCompanyInput: Locator;
  readonly contactWebsiteInput: Locator;
  readonly contactMessageTextarea: Locator;
  readonly contactSubmitButton: Locator;

  constructor(page: Page) {
    this.page = page;
    this.nav = new NavigationComponent(page);
    this.heroCtaButton = page.getByTestId("hero-cta-button");

    this.contactForm = page.getByTestId("contact-form");
    this.contactNameInput = page.getByTestId("contact-name-input");
    this.contactEmailInput = page.getByTestId("contact-email-input");
    this.contactCompanyInput = page.getByTestId("contact-company-input");
    this.contactWebsiteInput = page.getByTestId("contact-website-input");
    this.contactMessageTextarea = page.getByTestId("contact-message-textarea");
    this.contactSubmitButton = page.getByTestId("contact-submit-button");
  }

  async goto() {
    await this.page.goto("/");
  }

  async clickHeroCta() {
    await this.heroCtaButton.click();
  }

  async fillContactForm(data: {
    name: string;
    email: string;
    company?: string;
    website?: string;
    message: string;
  }) {
    await this.contactNameInput.fill(data.name);
    await this.contactEmailInput.fill(data.email);
    if (data.company) await this.contactCompanyInput.fill(data.company);
    if (data.website) await this.contactWebsiteInput.fill(data.website);
    await this.contactMessageTextarea.fill(data.message);
  }

  async submitContactForm() {
    await this.contactSubmitButton.click();
  }
}
